lib/Sema/CSRanking.cpp

```diff
@@ -1421,6 +1421,10 @@ ConstraintSystem::findBestSolution(SmallVectorImpl<Solution> &viable,
     }
   }
 
+  if (viable.size() == 2) {
+    return 0;
+  }
```

Comment:
so this is of course wrong, but I'm just hardcoding and trying to understand if we DID add the func + thunk to witnesses we could resolve it here, or if that's a completely wrong path towards making it work.